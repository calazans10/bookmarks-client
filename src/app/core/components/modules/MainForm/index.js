import React from 'react';
import PropTypes from 'prop-types';
import { Container, Legend } from './styles';

const MainForm = ({ legend, children, onSubmit }) => (
  <Container autoComplete="off" noValidate onSubmit={onSubmit}>
    <fieldset>
      <Legend>{legend}</Legend>
      {children}
    </fieldset>
  </Container>
);

MainForm.propTypes = {
  legend: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default MainForm;
