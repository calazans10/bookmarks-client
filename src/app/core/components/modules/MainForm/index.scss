@import 'styles/utilities';

.main-form {
  @media(min-width: 600px) {
    width: rem(522);
    margin: rem(60) auto;
  }

  @media(min-width: 800px) {
    margin-top: rem(100);
    margin-bottom: rem(100);
  }

  @media(min-width: 1044px) {
    width: rem(600);
  }

  &__legend {
    margin-bottom: rem(25);
    font-size: em(24);
    font-weight: 500;
  }

  a {
    margin-left: rem(10);
  }
}
