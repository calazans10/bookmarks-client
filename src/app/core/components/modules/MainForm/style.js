import styled from 'styled-components';
import { em, rem } from 'polished';

export const Container = styled.form`
  @media (min-width: 600px) {
    width: ${rem('522px')};
    margin: ${rem('60px')} auto;
  }

  @media (min-width: 800px) {
    margin-top: ${rem('100px')};
    margin-bottom: ${rem('100px')};
  }

  @media (min-width: 1044px) {
    width: ${rem('600px')};
  }

  a {
    margin-left: ${rem('10px')};
  }
`;

export const Legend = styled.legend`
  margin-bottom: ${rem('25px')};
  font-size: ${em('24px')};
  font-weight: 500;
`;
