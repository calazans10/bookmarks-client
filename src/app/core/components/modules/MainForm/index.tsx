import React from 'react';
import { Container, Legend } from './style';

type MainFormProps = {
  legend: string;
  children: React.ReactNode;
  onSubmit: () => void;
};

const MainForm = ({ legend, children, onSubmit }: MainFormProps) => (
  <Container autoComplete="off" noValidate onSubmit={onSubmit}>
    <fieldset>
      <Legend>{legend}</Legend>
      {children}
    </fieldset>
  </Container>
);

export default MainForm;
