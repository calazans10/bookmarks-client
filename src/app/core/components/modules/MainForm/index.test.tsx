import React from 'react';
import { render } from '@testing-library/react';
import MainForm from '.';

describe('MainForm', () => {
  it('renders without crashing', () => {
    const { container } = render(
      <MainForm legend="A legend" onSubmit={jest.fn()}>
        <div>
          <label htmlFor="test">A label</label>
          <input type="text" id="test" name="test" />
        </div>
      </MainForm>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
