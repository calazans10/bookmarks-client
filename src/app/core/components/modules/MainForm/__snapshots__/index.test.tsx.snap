// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainForm renders without crashing 1`] = `
<form
  autocomplete="off"
  class="sc-AxjAm hYYXIy"
  novalidate=""
>
  <fieldset>
    <legend
      class="sc-AxirZ dBtpcc"
    >
      A legend
    </legend>
    <div>
      <label
        for="test"
      >
        A label
      </label>
      <input
        id="test"
        name="test"
        type="text"
      />
    </div>
  </fieldset>
</form>
`;
